web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx

```diff
@@ -350,6 +376,121 @@ const CreateAccessRole = ({
   );
 };
 
+type TableData = {
+  items: SecurityGroup[];
+  nextToken?: string;
+  fetchStatus: FetchStatus;
+};
+
+const SelectSecurityGroups = ({
+  selectedSecurityGroups,
+  setSelectedSecurityGroups,
+  dbMeta,
+  emitErrorEvent,
+}: {
+  selectedSecurityGroups: string[];
+  setSelectedSecurityGroups(sg: string[]): void;
+  dbMeta: DbMeta;
+  emitErrorEvent(err: string): void;
+}) => {
+  const [sgTableData, setSgTableData] = useState<TableData>({
+    items: [],
+    nextToken: '',
+    fetchStatus: 'disabled',
+  });
+
+  const {
+    attempt: fetchSecurityGroupsAttempt,
+    setAttempt: setFetchSecurityGroupsAttempt,
+  } = useAttempt('processing');
+
+  function onSelectSecurityGroup(
+    sg: SecurityGroup,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) {
+    if (e.target.checked) {
+      return setSelectedSecurityGroups([...selectedSecurityGroups, sg.id]);
```

Comment:
Nit: when updating the state that depends on the previous state, we should use `setState` callback: https://dev.to/csituma/when-do-you-use-setstate-with-a-callback-1f3g